File: backend/src/models/DailyInventory.js
const mongoose = require('mongoose');

const dailyInventorySchema = new mongoose.Schema(
  {
    product: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Product',
      required: true,
    },
    productId: { type: String, required: true },
    farmer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    date: {
      type: String, // YYYY-MM-DD format
      required: true,
    },
    availableQuantity: {
      type: Number,
      required: true,
      min: [0, 'Available quantity cannot be negative'],
    },
    soldQuantity: {
      type: Number,
      default: 0,
      min: [0, 'Sold quantity cannot be negative'],
    },
    remainingQuantity: {
      type: Number,
      required: true,
      min: [0, 'Remaining quantity cannot be negative'],
    },
    unit: { type: String, required: true, default: 'L' },
    price: { type: Number, required: true },
    status: {
      type: String,
      enum: ['AVAILABLE', 'LIMITED', 'SOLD_OUT'],
      default: 'AVAILABLE',
    },
  },
  { timestamps: true }
);

// Ensure a single inventory document per product per farmer per day
dailyInventorySchema.index({ product: 1, farmer: 1, date: 1 }, { unique: true });

module.exports = mongoose.model('DailyInventory', dailyInventorySchema);

File: backend-dotnet/Models/DailyInventory.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DailyMarts.Api.Models;

[BsonIgnoreExtraElements]
public class DailyInventory
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonElement("product")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Product { get; set; } = string.Empty;

    [BsonElement("productId")]
    public string ProductId { get; set; } = string.Empty;

    [BsonElement("farmer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Farmer { get; set; } = string.Empty;

    [BsonElement("date")]
    public string Date { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-dd");

    [BsonElement("availableQuantity")]
    public double AvailableQuantity { get; set; }

    [BsonElement("soldQuantity")]
    public double SoldQuantity { get; set; } = 0;

    [BsonElement("remainingQuantity")]
    public double RemainingQuantity { get; set; }

    [BsonElement("unit")]
    public string Unit { get; set; } = "L";

    [BsonElement("price")]
    public decimal Price { get; set; }

    [BsonElement("status")]
    public string Status { get; set; } = "AVAILABLE"; // AVAILABLE, LIMITED, SOLD_OUT

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [BsonIgnore]
    public Product? ProductDetails { get; set; }

    [BsonIgnore]
    public User? FarmerDetails { get; set; }
}
